doc/sphinx/language/core/basic.rst

```diff
@@ -437,6 +437,43 @@ boldface label "Attribute:".  Attributes are listed in the
       Set Warnings "unsupported-attributes".
       #[ foo ] Comments.
 
+Generic attributes
+^^^^^^^^^^^^^^^^^^
+
+The following attribute is supported by every command:
+
+.. attr:: warnings = @string
+   :name: warnings
+
+   Sets the given warning string locally for the command. After the
+   command finishes the warning state is reset to what it was before
+   the command. For instance if the current warning state is
+   `some-warnings,-other-warning`,
+
+   .. coqdoc::
+
+      #[warnings="+other-warning"] Command.
+
+   is equivalent to
+
+   .. coqdoc::
+
+      Set Warnings "+other-warning".
+      Command.
+      Set Warnings "some-warnings,-other-warning".
```

Comment:
Is some-warnings really needed?  IIUC unmentioned warnings are not changed and simpler for the reader to understand.

Reply:
OK, but beyond the scope of this PR:  Does `Set Warnings "+my-warnings"` reset _all_ warnings?  The doc does not cover this detail.  If so, that seems unfriendly: if you have multiple cumulative `Set Warnings`, the second command has to repeat the settings from the first command (and similarly for the 3rd, ...)  IIUC this is different from how OCaml enables/disables warnings.

The doc for `Warnings` says + to make a warning an error and - to disable the warning.  Maybe that should be something like ! to make a warning an error, + to enable display of the warning (as a warning and not as an error) and - to disable displaying the warning.  And an entry without a prefix resets all warnings (if that's actually useful).

Reply:
>beyond the scope of this PR

answered in zulip